gto/constants.py

```diff
@@ -42,7 +42,7 @@ def tag_to_name(value):
 
 
 dirname = "[a-z0-9-_./]+"  # improve?
-name = r"[a-z0-9]([a-z0-9-/]*[a-z0-9])?"
+name = r"[A-Za-z0-9]([A-Za-z0-9-/_]*[A-Za-z0-9])?"
```

Comment:
not sure about modern Git, but In general the recommendation is to avoid using upper case tag / branch names (since Windows and Mac could mess up with that). Let's just add some symbols for now? Did we see people complaining? We could also automatically convert for simplicity / usability when people manipulate / pass those names in UI / CLI.

(same btw applies to the semver regex that I see is also using `A-Z`

wdyt?

Reply:
No strong opinion from me (Git does handle casing from what research I did but it's obviously not the typical convention). @mnrozhkov You had asked about this, right? Do you find it useful/common to need uppercase letters?

Reply:
+1 to Ivan's comment here. I think the simpler it is, the less changes to make a mistake (like mistake `i`.upper() from `L`.lower()), etc. That was original intention.

Reply:
> No strong opinion from me (Git does handle casing from what research I did but it's obviously not the typical convention). @mnrozhkov You had asked about this, right? Do you find it useful/common to need uppercase letters?

I think we should support uppercase letters for artifact names! 

[Here](https://iterativeai.slack.com/archives/CB41NAL8H/p1691411376466919) I raised a discussion because of an error by calling `live.log_artifact("scalarTransportFoam")` 
- The "scalarTransportFoam" is a directory name generated by third-party software (simulation). There is no option to name it by the DVC user

Reply:
@mnrozhkov Do you think it's important to keep the casing, or do you think the suggestion to force the name to lowercase would be acceptable?

Reply:
@dberenbaum as I answered above: 
> I think we should support uppercase letters for artifact names!

It's not acceptable for the customer I mentioned to change the name to lowercase 
1. Because this name is generated by a third-party software, it's standard 
2. This name may be used by the same software or legacy code, so it could break the workflow 

I feel it's fine to keep lowercase letters for git tag/branch things, but should we be so strict for artifact names? 
Feel free to have a call if I miss smth here.

Reply:
Do others feel it's a blocker? I don't see a strong reason not to allow it even if it's not the typical recommended pattern.

Reply:
> I feel it's fine to keep lowercase letters for git tag/branch things, but should we be so strict for artifact names?
Feel free to have a call if I miss smth here.

So, is it fine to "complain" then if someone is trying to register an artifact with a different name?

I'm fine to keep the change, also seems reasonable to allow this on the dvc.yaml level. I would still probably convert this into lowercase when we do tags. Just to be safe here.

Reply:
It's going to expand the scope here to have special logic for how to handle uppercase letters, so I'm going to drop that from the PR and let's keep a separate issue to discuss it.